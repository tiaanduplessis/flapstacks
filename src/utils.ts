export const removeUndefined = (obj: { [key: string]: any }) => {
  Object.keys(obj).forEach((key) => {
    if (obj[key] === undefined) delete obj[key];
  });
  return obj;
};

export const when = (value: any, trueValue?: string, falseValue?: string) => {
  if (trueValue && value === true) return trueValue;
  if (falseValue && value === false) return falseValue;
  return value;
};

export const match = (...values: any[]) => {
  for (let value of values) {
    if (value !== undefined) return value;
  }
};
