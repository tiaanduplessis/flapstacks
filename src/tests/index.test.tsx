import React, { CSSProperties } from 'react'
import { describe, test } from 'vitest'
import {Stack} from '../'
import { matchesSnapshot } from './utils'

describe('Stack', () => {
  test('should render Stack with defaults', () => {
    matchesSnapshot(<Stack />)
  })

  test('should allow using style prop', () => {
    matchesSnapshot(<Stack style={{ color: 'red' }} />)
  })

  test('should be polymorhic component', () => {
    matchesSnapshot(<Stack as="main"/>)
    matchesSnapshot(<Stack component="footer"/>)
  })

  test('should render Stack as inline', () => {
    matchesSnapshot(<Stack inline />)
  })

  test('should render Stack as row', () => {
    matchesSnapshot(<Stack />)
    matchesSnapshot(<Stack direction='row' />)
    matchesSnapshot(<Stack flexDirection='row' />)
    matchesSnapshot(<Stack row />)
  })

  test('should render Stack as column', () => {
    matchesSnapshot(<Stack flexDirection='column' />)
    matchesSnapshot(<Stack direction='column' />)
    matchesSnapshot(<Stack column />)
  })

  test('should render with gap',() => {
    matchesSnapshot(<Stack gap='8px' />)
    matchesSnapshot(<Stack columnGap='8px'/>)
    matchesSnapshot(<Stack rowGap='8px'/>)
  })

  test('should render with justified content',() => {
    matchesSnapshot(<Stack main="flex-start" />)
    matchesSnapshot(<Stack justify="flex-end" />)
    matchesSnapshot(<Stack justifyContent="center" />)
    matchesSnapshot(<Stack justifyContentCenter />)
    matchesSnapshot(<Stack justifyContentStart />)
    matchesSnapshot(<Stack justifyContentEnd/>)
    matchesSnapshot(<Stack justifyContentSpaceBetween/>)
    matchesSnapshot(<Stack justifyContentSpaceAround/>)
  })

  test('should allow overriding styles', () => {
    const handleOverride = (style: CSSProperties) => {
      const scale = ["16px", "32px"]
      if (style.gap) {
        style.gap = scale[style.gap as number];
      }
    
      return style;
    }
    matchesSnapshot(<Stack gap={1} onOverrideStyles={handleOverride}/>)
  })
})