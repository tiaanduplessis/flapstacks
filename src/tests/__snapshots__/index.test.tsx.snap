// Vitest Snapshot v1

exports[`Stack > should allow for gap prop 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "8px",
    }
  }
/>
`;

exports[`Stack > should allow for gap prop 2`] = `
<div
  style={
    {
      "columnGap": "8px",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should allow for gap prop 3`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "rowGap": "8px",
    }
  }
/>
`;

exports[`Stack > should allow overriding styles 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "32px",
    }
  }
/>
`;

exports[`Stack > should allow using style prop 1`] = `
<div
  style={
    {
      "color": "red",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should be polymorhic component 1`] = `
<main
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should be polymorhic component 2`] = `
<footer
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as column 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as column 2`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as column 3`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as inline 1`] = `
<div
  style={
    {
      "display": "inline-flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as row 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as row 2`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as row 3`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack as row 4`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;

exports[`Stack > should render Stack with defaults 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
/>
`;
